### ISSUE TITLE
Networkpolicy causing schism between cli and dashboard

### ISSUE THREAD

**Issue by oivindoh** [linkerd/linkerd2#864]
**TL;DR version: network policies caused dashboard to report deployments targeted by these policies as not in mesh, while the conduit cli was of another opinion. Allowing the conduit namespace to contact the pods in question solves the issue.**

After I deployed conduit 0.4.1 using `conduit install | kubectl apply -f -` in our fresh cluster, looked at the emojivoto example to see what had changed to deployment targeting/sidecar setup since 0.3.1, and made the necessary changes to my deployments, I expected them to end up in `conduit dashboard`, but was sad to see proxy status "not in mesh" for my pods. Looking at the proxy logs for the pods I couldn't find any issue:

```
INFO conduit_proxy using controller at HostAndPort { host: DnsName(Name("proxy-api.conduit.svc.cluster.local")), port: 8086 }
INFO conduit_proxy routing on V4(127.0.0.1:4140)
INFO conduit_proxy proxying on V4(0.0.0.0:4143) to None
INFO conduit_proxy serving Prometheus metrics on V4(0.0.0.0:4191)
INFO conduit_proxy protocol detection disabled for inbound ports {25, 3306}
INFO conduit_proxy protocol detection disabled for outbound ports {25, 3306}
INFO conduit_proxy::transport::connect "controller-client", DNS resolved DnsName(Name("proxy-api.conduit.svc.cluster.local")) to 10.0.239.98 
```

The cli was of another opinion than the dashboard, however:

```
$ conduit stat namespaces
NAME                MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99
oho                    5/5         -        -             -             -             -
conduit                4/4    82.48%   2.3rps           3ms          48ms         176ms
emojivoto              4/4    90.17%   4.9rps           1ms          12ms          25ms

$ conduit stat deployments -n oho
NAME                       MESHED   SUCCESS   RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99
frontend-api                  1/1         -     -             -             -             -
frontend-ml                   1/1         -     -             -             -             -
frontend-onboarding           1/1         -     -             -             -             -
frontend-redis                1/1         -     -             -             -             -
frontend-web                  1/1         -     -             -             -             -
```

After some head scratching I remembered that the pods in these deployments are targeted by fairly strict network policies (using quay.io/calico/node:v2.6.3 as the provider). Here's an example of an offending namespace that only allows traffic from two other aptly labelled namespaces:

```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: in-policy
spec:
  podSelector:
    matchLabels:
      app: frontend
  policyTypes:
  - Ingress
  ingress:
    - from:
      - namespaceSelector:
          matchLabels:
            purpose: ingress
      - namespaceSelector:
          matchLabels:
            purpose: monitoring
```

(correctly) labeling the conduit namespace with the `purpose: monitoring` label allowed the deployments in all my namespaces to register with conduit as expected, and I'm now busy rather enjoying the wonderful grafana dashboards you've made 👍
